tree-sitter/src/TreeSitter/Span.hs

```diff
@@ -0,0 +1,18 @@
+module TreeSitter.Span
+( Span(..)
+, Pos(..)
+) where
+
+import Prelude hiding (span)
+
+-- | Source position information
+data Pos = Pos
+  { posLine   :: {-# UNPACK #-} !Int
+  , posColumn :: {-# UNPACK #-} !Int
+  } deriving (Eq, Ord, Show)
+
+-- | A Span of position information
+data Span = Span
+  { spanStart :: {-# UNPACK #-} !Pos
+  , spanEnd   :: {-# UNPACK #-} !Pos
```

Comment:
I copied `Range`/`Span` definitions from Semantic's `Data.Range` and `Data.Span`. I noticed `{-# OPTIONS_GHC -funbox-strict-fields #-}` is used over in Semantic, although `UNPACK` seems to achieve the same thing. 

Confirmed with @patrickt that `-funbox-strict-fields` is like applying `UNPACK` pragmas to every strict field. @robrix clarified that Semantic uses  StrictData project-wide, so `-funbox-strict-fields` means every unpackable field in that file is unpacked.

Theoretically, I could remove `UNPACK` and just use the pragma, since all fields are strict and now they’re in their own module, but @robrix pointed out that's probably not ideal since I’d be relying on the reader knowing & remembering to look for the `OPTIONS_GHC` pragma, so I've left it as it is!

Reply:
This looks good. Let’s be sure to bump versions to 0.2.1.0 and let’s make a note to switch to using these inside semantic proper.

Reply:
`semantic` should probably continue using its own definitions (with the addition of `Unmarshal` instances), or else several of the instances it defines would be orphaned.

Reply:
I think orphans would be better than paying a per-node cost of unpacking/repacking identical span/range structures, but this is all hypothetical. We’ll cross that 🌉 when we come to it!

Reply:
We’ll unmarshal directly into the destination datatypes via _their_ instances, so no such cost will be incurred.